examples/lib/cache.js

```diff
@@ -29,24 +29,19 @@ var setupCache = function() {
                 if(navigator.serviceWorker.controller) {
                   // New service worker available; prompt the user to update.
                   showUpdateModal();
+                  $('#reload').on('click',(e) => {
+                    e.preventDefault();
+                    console.log('New Service Worker Installed Successfully');
+                    location.reload();
+                  })
                 }
                 // No updates available; do nothing.
                 break;
             }
-          });
-        });
-
-        const installingWorker = registration.installing;
```

Comment:
this is happening bcoz we have already checked service worker state as installed and now this state `installing` will  not come again. so it returns a null on which we have just added a `statechange` event listener which is meaningless.

Reply:
I see, thank you!; this is so hard to track as maintainers; do you think it's possible to write a test for this?

Reply:
I have tried to write test for this but i don't know how to make this run 😅  ? 

you can have a look this test -
https://github.com/vivek-30/image-sequencer/blob/5fc7fc8ecbb02ac989605a4d422b7b8d85d67aeb/test/core/sw.test.js#L1-L72

@jywarren, @HarshKhandeparkar  could you please suggest me what to do?

Reply:
Actually @jywarren  what i think is even if we don't build a test for this then also it will solve this problem.
No doubt that adding test will ensure more security and safety but don't think it will slow down image sequencer as to check a successful installation of sw .we have to remove existing sw and clear out the cache and then register a new updated sw (which i have done in my test) ?

Reply:
Thanks for giving it a try! If you add the test to the correct folder alongside others in this repo, it'll run within this PR. It's possible you need to add some more assertions -- like, see how this line confirms something we expect to be true about the test scenario:

https://github.com/publiclab/image-sequencer/blob/eb81b7de4b331dc32684e90ee9898520c80c14e4/test/core/sequencer/chain.js#L16

Want to try adding your test in this PR?

Reply:
@jywarren plz have a look now ,i have added a test for this. Hope you like it ☺️